extensions/2.0/Vendor/GRIFFEL_bim_data/schema/glTF.GRIFFEL_bim_data.schema.json

```diff
@@ -0,0 +1,58 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "title": "GRIFFEL_bim_data glTF Extension",
+    "type": "object",
+    "description": "Domain specific properties for glTF nodes.",
+    "allOf": [ { "$ref": "glTFProperty.schema.json" } ],
+    "properties": {
+        "nodeProperties": {
+            "type": "array",
+            "items": {
+                "$ref": "node.properties.schema.json",
+                "description": "Node properties and type."
+            },
+            "description": "Node type and instance properties map. Used only when data is saved in a separate file."
+        },
+        "propertyValues": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "description": "Property value.",
+                "minLength": 1
+            },
+            "minItems": 1,
+            "description": "Collection of unique property values."
+        },
+        "propertyNames": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "description": "Property name.",
+                "minLength": 1
+            },
+            "minItems": 1,
+            "description": "Collection of unique property names."
+        },
+        "properties": {
+            "type": "array",
+            "items": {
+                "$ref": "property.schema.json",
+                "description": "Node property."
+            },
+            "minItems": 1,
+            "description": "Collection of unique property name - property value pairs."
+        },
+        "types": {
+            "type": "array",
+            "items": {
+                "$ref": "type.schema.json",
```

Comment:
The `type.schema.json` is quite simple and used only once. so it should be just embedded here.

Reply:
Sorry, this is _somewhat_ unrelated to this specific PR, but I stumbled over this: When wetzel is supposed to be used for generating a schema reference (for example, for this extension), then it may be crucial to give these things a _name_ (the name being `type` here). Otherwise, it can be hard or impossible to generate the

> An array of types.
> 
> - Type: [type](http://example.com) [1-*]

part with the link to the type.

Reply:
Embedded type.schema, thou didn't got the @javagl point.

Reply:
@ibober That may have been too unrelated without some context, but the [properties reference](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#properties-reference) of glTF and related specifications is created with the [wetzel](https://github.com/CesiumGS/wetzel) tool. And this tool will have difficulties listing the information for such "embedded" schemas: It will no longer be able to create a proper list of types or the properties, or a link to the documentation of this schema. 

@lexaknyazev More generally, I wondered whether it could make sense to carve out some "best practices" for JSON schema writing _in general_, as part of the guidelines for creating extensions. There are many things that _can_ be modeled in JSON schema, but are not as clean or idiomatic as the core of glTF, and that may cause unnecessary hassle on the side of possible implementors (including the validator).